tests/functional/metric_options/where/test_where_expression_metric.py

```diff
@@ -0,0 +1,133 @@
+from struct import pack
+import os
+import pytest
+from dbt.tests.util import run_dbt
+
+# our file contents
+from tests.functional.fixtures import (
+    fact_orders_source_csv,
+    fact_orders_sql,
+    fact_orders_yml,
+)
+
+# models/where_expression_metric.sql
+where_expression_metric_sql = """
+select *
+from 
+{{ metrics.calculate(metric('where_expression_metric'), 
+    grain='month',
+    dimensions=['had_discount'],
+    where=["had_discount=true"]
```

Comment:
this is interesting! I don't hate it, but it is different to every other package's `where` strategy ([example](https://github.com/dbt-labs/dbt-utils#accepted_range-source)), which is "just give us a whole string and we'll whack it in" as opposed to an array of individual strings where the package takes responsibility for putting the `and`s in. 

I'm not averse to it, but wanted to call out that it's nontraditional

Reply:
You know, I'd gone with the list format because it matched dimensions and our original thinking was allowing BI tools to input small snippets of filters based on user inputs. 

BUT, the more I think about it, the more this is a pretty inflexible method of implementation. If there are multiple filters, are they default AND? What if I want 2 ANDs and 1 OR? Or group two of the filters together? It is beginning to remind me of how we implemented filters at a last job and that ended up having more issues than benefits. 

I'm going to switch it over to the more traditional `where` to match dbt_utils